Cython/Compiler/ModuleNode.py

```diff
@@ -1628,6 +1629,27 @@ def generate_new_function(self, scope, code, cclass_entry):
         code.putln(
             "}")
 
+    def generate_del_function(self, scope, code):
+        tp_slot = TypeSlots.get_slot_by_name("tp_finalize", scope.directives)
+        slot_func_cname = scope.mangle_internal("tp_finalize")
+        if tp_slot.slot_code(scope) != slot_func_cname:
+            return  # never used
+
+        entry = scope.lookup_here("__del__")
+        if entry is None or not entry.is_special:
+            return  # nothing to wrap
+        slot_func_cname = scope.mangle_internal("tp_finalize")
+        code.putln("")
+
+        code.putln("#if !CYTHON_COMPILING_IN_LIMITED_API")
```

Comment:
Shouldn't this condition be aligned with the `used_ifdef` below? That uses `CYTHON_USE_TP_FINALIZE`. However, the latter is also enabled for the limited API. That seems conflicting.

Reply:
Yes I think so. It gives a compile error in the limited API as it is, but works with `#if CYTHON_USE_TP_FINALIZE`